src/v/storage/index_state.cc

```diff
@@ -429,24 +567,26 @@ index_state index_state_serde::decode(iobuf_parser& parser) {
     const uint32_t vsize = ss::le_to_cpu(
       reflection::adl<uint32_t>{}.from(parser));
 
+    chunked_vector<uint32_t> relative_offset_index;
+    chunked_vector<uint32_t> relative_time_index;
+    chunked_vector<uint64_t> position_index;
```

Comment:
Is there a reason we don't call `reserve(vsize)` on each of these?

Reply:
there is no good reason, I just converted the existing code mechanically but it will not hurt to reserve memory